### ISSUE TITLE
Running Rancher Desktop returns [process exited with code 1 (0x00000001)] 

### ISSUE THREAD

**Issue by micheleietri** [rancher-sandbox/rancher-desktop#1919]
Hi,

I'm trying to install rancher desktop 1.2.1 on my PC (Windows 11 22H2 22581.100, insider preview)

I've downloaded 1.2.1 version intaller from github, double click to run it (tried even "as administrator").
After a while installation wizard pops up, and I can reach last page with no error prompted to end installation process and run the app, 
but after a while (10/20 seconds) a terminal window opens with the error message [process exited with code 1 (0x00000001)].

No other errors nor logs.

But installing this way if I check wsl instances (wsl --list) I can see just my ubuntu one.

I tried to install 1.0.0 version, and things are slightly different.

same process as described before, wizard completes without any error, but when I click finish after a while same error appears, but then init wizard appears (where i can set k8s version, container runtime etc) and with wsl --list it shows rancher-desktop and rancher-desktop-data distributions on my system.

but after that if i upgrade same issue as before, terminal widows with error, even if i can see those 2 distributions

any suggestions?

**Comment by gunamata** [comment]
Do you see any logs generated at `%USERPROFILE%\AppData\Local\rancher-desktop\logs` when the installation fails? If yes, Can you please share the same with us?

Also, the name of the application is `Rancher Desktop`  😊

**Comment by micheleietri** [comment]
Sorry, (can be considered typo?)

by the way, the logs folders contains those files

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----        29/03/2022     17:10           1853 [background.log](https://github.com/rancher-sandbox/rancher-desktop/files/8372775/background.log)
-a----        29/03/2022     17:10              0 images.log
-a----        29/03/2022     17:10            783 [k8s.log](https://github.com/rancher-sandbox/rancher-desktop/files/8372776/k8s.log)
-a----        29/03/2022     17:10              0 lima.log
-a----        29/03/2022     17:10             51 [server.log](https://github.com/rancher-sandbox/rancher-desktop/files/8372777/server.log)
-a----        29/03/2022     17:10              0 settings.log
-a----        29/03/2022     17:10              0 steve.log
-a----        29/03/2022     17:10              0 update.log
-a----        29/03/2022     17:10              0 wsl.log

Still no clue on what's happening :/
I've already seen suggestion in issue #1072 and tried [this](https://github.com/rancher-sandbox/rancher-desktop/issues/1072#issuecomment-1046913704) fix, but still nothing changed.

**Comment by micheleietri** [comment]
Other colleagues with same schema in username had no problem intstalling and running rancher

**Comment by mook-as** [comment]
`background.log` has the same `roots.exe` reporting `ENOBUFS`; that sounds like #1072 (which should be fixed in the next release).

**Comment by MarvinEads** [comment]
Fixed in next release? K, I know release 1.2.1 is working for a co-worker after his broke but mine is still breaking the same was as 1072 said even after updating to 1.2.1.

**Comment by mook-as** [comment]
If you'd like to help us confirm that it's fixed, there's always the [CI builds](https://github.com/rancher-sandbox/rancher-desktop/actions/workflows/package.yaml?query=branch%3Amain) — just grab the newest files from `main` and run it to check that it works :)

**Comment by MarvinEads** [comment]
Oh fine, just call me out and make me self sufficient...  :)

Downloaded latest installed for Win and it installed and opened. Waiting for Kuber to install now, got an error starting but assuming its cause it needed stuff to DL first. Will update after

**Comment by flexisz** [comment]
rancher desktop basically starts, but then gets an error when trying to start wsl

```
ln -s -f -T "/mnt/c/Users/x.x/AppData/Roaming/rancher-desktop/provisioning" /etc/local.d

          # Ensure all scripts are executable; Windows mounts are unlikely to
          # have it set by default.
          /usr/bin/find             /etc/local.d/             '(' -name '*.start' -o -name '*.stop' ')'             -print -exec chmod a+x '{}' ';'

          # Run the script.
          exec /usr/local/bin/wsl-service local start
        : Error: wsl.exe exited with code 1
```

or

`wsl.exe --distribution rancher-desktop --exec /bin/sh -c set -o errexit -o nounset # Stop the service if it's already running for some reason. # This should never be the case (because we tore down init). /usr/local/bin/wsl-service --ifstarted local stop # Clobber /etc/local.d and replace it with a symlink to our desired # path. This is needed as /etc/init.d/local does not support # overriding the script directory. rm -r -f /etc/local.d ln -s -f -T "/mnt/c/Users/x.x/AppData/Roaming/rancher-desktop/provisioning" /etc/local.d # Ensure all scripts are executable; Windows mounts are unlikely to # have it set by default. /usr/bin/find /etc/local.d/ '(' -name '*.start' -o -name '*.stop' ')' -print -exec chmod a+x '{}' ';' # Run the script. exec /usr/local/bin/wsl-service local start`

**Comment by MarvinEads** [comment]
Pretty much same as above. Did some research and lot of references to that WSL code being related to something listening on port 53 but I am not showing that condition. Someone also mentioned their client VPN app installed some DNS lookup application too, recently changed VPN providers as work but this was working with that so dont think its related. Hoping to get some time to look more today.

Error: wsl.exe exited with code 4294967295

**Comment by CuriousWizard** [comment]
> Error: wsl.exe exited with code 4294967295
If you happen to see this error message, I usually do a "Reset Kubernetes" under Kubernetes Settings and it solves this error. ([issue](https://github.com/rancher-sandbox/rancher-desktop/issues/1786))

**Comment by MarvinEads** [comment]
Resetting results in the below loglines and err:

// checked the referenced directory, its empty, maybe a timing issue or something?
Error: ENOTEMPTY: directory not empty, rmdir 'C:\Users\[USER]\AppData\Local\Temp\rd-ca-2qQQF9'

2022-04-04T12:36:01.933Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-389.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:01.963Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-387.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:01.973Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-384.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:01.975Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-390.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:01.986Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-75.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:02.004Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-388.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:02.029Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-291.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:02.047Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-391.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1
2022-04-04T12:36:02.088Z: WSL: executing: cp /mnt/c/Users/[USER]/AppData/Local/Temp/rd-ca-2qQQF9/rd-187.crt /usr/local/share/ca-certificates/: Error: wsl.exe exited with code 1

**Comment by flexisz** [comment]
the same for me

**Comment by vladonemo** [comment]
I'm facing the same. Installed the latest version today:
https://github.com/rancher-sandbox/rancher-desktop/actions/runs/2192974373


I believe that it is a timing issue. The `rd-ca-*` temp folder is deleted before all the certs are copied to the WLS instance. 
Attaching my wsl.log
[wsl.log](https://github.com/rancher-sandbox/rancher-desktop/files/8520416/wsl.log)

The folder doesn't exist any longer and yet there are hundreds of parallel processes trying to copy the certs over:

![image](https://user-images.githubusercontent.com/2752296/164202178-c52269ae-9f6c-4837-88d8-58777b65490b.png)

**Comment by vladonemo** [comment]
I have spent a decent amount of time debugging this. The PR that I just submitted fixes the issue related to certificates. In my case there are almost 500 certificates to be installed. 2 separate `wsl` processes are spawned for each certificate (one to wlsify the path and one to copy the file). If any of these fail on exception (timeout?), then the workdir is removed with all the certs yet to be copied over. All the remaining processes will fail (after some timeout, that can sum up to long minutes).

With the PR I was able to get a bit further. But I'm facing the issue that @flexisz described in his [comment](https://github.com/rancher-sandbox/rancher-desktop/issues/1919#issuecomment-1084452881). For some reason both the `/etc/rancher` and `/var/lib` are not mounted. Hence the `wsl-init` fail when calling `mount --make-shared /etc/rancher`. You can find the error message saying that `/sbin/init` failed, then the provisioning script fails as well.

**Comment by vladonemo** [comment]
I believe that I have found a patch for the issue with failing `wsl-init`. It really boils down to the fact that there are actions running on WSL in parallel. The fix is to await the `wsl-init` to finish before moving on with installing trivy, certificates, helpers and k3s.
A separate PR will follow.
